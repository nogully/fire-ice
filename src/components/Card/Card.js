import React from 'react';

export const Card = ({ house, handleClick }) => {
  const { name, founded, seats, titles, coatOfArms, ancestralWeapons, words, swornMembers } = house; 
  const swornMemberElements = swornMembers.map(member => {
    if (member.name){
        return (
          <p>{member.name} : {member.status} </p>
        )
    } else {return null}
  })
  
  return (
    <article className="Card" id={name} onClick={handleClick}>
      <h3>Name: {name}</h3>
      <p>Founded: {founded}</p>
      <p>Seats: {seats}</p>
      <p>Titles: {titles}</p>
      <p>Coat of Arms: {coatOfArms}</p>
      <p>Ancestral Weapons: {ancestralWeapons}</p>
      <p>{words}</p>
      { swornMemberElements }
    </article> 
  )
}

export default Card;