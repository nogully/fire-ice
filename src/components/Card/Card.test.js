import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card'

describe('Card', () => {
  const mockHouse = { name: "House Dayne of Starfall", 
                      swornMembers: []}

  it('should look like its snapshot', () => {
    const wrapper = shallow(<Card house={mockHouse} />)
    expect(wrapper).toMatchSnapshot();
  })
})