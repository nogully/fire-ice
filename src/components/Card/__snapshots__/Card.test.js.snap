// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should look like its snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    house={
        Object {
            "name": "House Dayne of Starfall",
            "swornMembers": Array [],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Name: 
          House Dayne of Starfall
</h3>,
        <p>
          Founded: 
</p>,
        <p>
          Seats: 
</p>,
        <p>
          Titles: 
</p>,
        <p>
          Coat of Arms: 
</p>,
        <p>
          Ancestral Weapons: 
</p>,
        <p />,
        Array [],
      ],
      "className": "Card",
      "id": "House Dayne of Starfall",
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name: ",
            "House Dayne of Starfall",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Name: ",
          "House Dayne of Starfall",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Founded: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Founded: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Seats: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Seats: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Titles: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Titles: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Coat of Arms: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Coat of Arms: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ancestral Weapons: ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Ancestral Weapons: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Name: 
            House Dayne of Starfall
</h3>,
          <p>
            Founded: 
</p>,
          <p>
            Seats: 
</p>,
          <p>
            Titles: 
</p>,
          <p>
            Coat of Arms: 
</p>,
          <p>
            Ancestral Weapons: 
</p>,
          <p />,
          Array [],
        ],
        "className": "Card",
        "id": "House Dayne of Starfall",
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name: ",
              "House Dayne of Starfall",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Name: ",
            "House Dayne of Starfall",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Founded: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Founded: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Seats: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Seats: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Titles: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Titles: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Coat of Arms: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Coat of Arms: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ancestral Weapons: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Ancestral Weapons: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
